### PR TITLE
docs: add contributing.md

### LINKED ISSUE #30: Add Contributing Doc
## Summary

Add a contributing document to the root of the repo. People interested in contributing for the first time can use this document to understand how they should get started. For a basic example, please refer to the [contributing docs in the js contrib repo](https://github.com/open-feature/js-sdk-contrib/blob/main/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by Sindhuinti**
## This PR
This PR adds a CONTRIBUTING.md file to the project, which provides guidelines for contributors on how to contribute to the project. The CONTRIBUTING.md file is a common best practice for open-source projects, and helps to ensure that contributors are aware of the project's guidelines and expectations.

## Changes Made

Added CONTRIBUTING.md file to the root directory of the project.
Included information on adding new module  and testing requirements.
Included instructions on how to set up the development environment and run the project locally.
## Why is this important?
Adding a CONTRIBUTING.md file helps to ensure that contributors are aware of the project's guidelines and expectations. This can help to reduce confusion and increase the likelihood of high-quality contributions to the project.

Additionally, having a CONTRIBUTING.md file can help to make the project more accessible to new contributors, as it provides a clear set of guidelines and expectations for how to contribute.

### Related Issues
This PR addresses issue #30 , which requested the addition of a CONTRIBUTING.md file to the project.
Fixes #30

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+make workspace-init
+```
+
+sync go.work with current modules
+
+```
+make workspace-update
+```
+
+## Compilation target(s)
```
Comment (Kavindu-Dodan) [inline CONTRIBUTING.md:20]: This section is not applicable for go-sdk as we do not build artifacts. Instead, SDK relies on Go lang's source-based dependencies. 

Note - See below example for clarification :) 

```
package main

import (
	"context"
	"fmt"
	"github.com/open-feature/go-sdk-contrib/providers/flagd/pkg"
	"github.com/open-feature/go-sdk/pkg/openfeature"
)

func main() {
	openfeature.SetProvider(flagd.NewProvider())
	client := openfeature.NewClient("app")
	value, _ := client.BooleanValue(
		context.Background(), "v2_enabled", false, openfeature.EvaluationContext{},
	)

	fmt.Printf("flag eval: %t", value)
}

```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Testing packages provide shared testing functionality across OpenFeature components, avoiding duplication.
+
+To test all go modules
+```
+make test
+```
+
+## Releases
+
+This repo uses _Release Please_ to release packages. Release Please sets up a running PR that tracks all changes for the library components, and maintains the versions according to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), generated when [PRs are merged](https://github.com/amannn/action-semantic-pull-request). When Release Please's running PR is merged, any changed artifacts are published.
```
Comment (Kavindu-Dodan) [inline CONTRIBUTING.md:65]:
```suggestion
This repo uses _Release Please_ to release packages. Release Please sets up a running PR that tracks all changes for the library components, and maintains the versions according to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), generated when [PRs are merged](https://github.com/amannn/action-semantic-pull-request).

Merging the Release Please PR will create a GitHub release with updated library versions.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## Dependencies
+
+Keep dependencies to a minimum, especially non-dev dependencies.
+The GO-SDK should be a _peer dependency_ of your module.
+Run `go mod`, and then verify the dependencies in `go.mod` file are appropriate.
+Keep in mind, though one version of the GO-SDK is used for all modules in testing, each module may have a different peer-dependency requirement for the GO-SDK (e.g: one module may require ^1.18.0 while another might require ^1.19.0).
```
Comment (Kavindu-Dodan) [inline CONTRIBUTING.md:72]: This is not applicable as individual modules can have their own compatible go-sdk versions and it works here :)

**Comment by Kavindu-Dodan** [comment]
@Sindhuinti thank you for the contribution. 

I added comments on where we need to improve the contribution guide. Most of them are related to how Go lang works (compared to Java, Js). I hope you will correct them as well as it helps you to learn new concepts

**Comment by Sindhuinti** [comment]
Hey @Kavindu-Dodan Thanks for the review. I learnt new things regrading go-sdk.
Please check the PR changes.

**Review by Kavindu-Dodan — APPROVED** [review]

**Comment by Kavindu-Dodan** [comment]
> Hey @Kavindu-Dodan Thanks for the review. I learnt new things regrading go-sdk. Please check the PR changes.

Nice work. Approved :)

**Review by beeme1mr — APPROVED** [review]
Thank you!

**Pull request merged**